server/src/test/java/org/elasticsearch/snapshots/DedicatedClusterSnapshotRestoreIT.java

```diff
@@ -497,11 +497,10 @@ public void testSnapshotWithStuckNode() throws Exception {
         logger.info("--> Go through a loop of creating and deleting a snapshot to trigger repository cleanup");
         client().admin().cluster().prepareCleanupRepository("test-repo").get();
 
-        // Subtract four files that will remain in the repository:
+        // Expect two files to remain in the repository:
```

Comment:
This whole business of keeping a "backup" index-N has long been fairly obsolete. At best it may have saved some repository status API calls from erros on S3 (when the most recent `index-N` would not show up in a listing ... but obviously then also serving old state anyway). With a consistent way of tracking the generation for writes now, I don't see any point in keeping this behavior around and simplified the deleting of old `index-N` accordingly in production code.